Mathlib/Lean/Expr/Lit.lean

```diff
@@ -0,0 +1,55 @@
+/-
+Copyright (c) 2025 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Qq
+
+/-!
+# Extract literals from expressions
+
+This file provides extra utility function to extract raw natural and integer literals from
+expressions.
+-/
+
+open Lean
+
+namespace Lean.Expr
+
+/-- Returns the integer from a raw integer literal expression, or `none` if the expression isn't
+a raw integer literal.
```

Comment:
I claim this is not something you ever need to do. "Raw integer literal expression" is a concept from only within `norm_num`, where you get handed the cases separately anyway.

Reply:
"this" = "using this function"? Then what should I replace it with in the Qq function?

Reply:
Paul wanted it to extract`10` from `Int.divisorsAntidiag 10`

Reply:
There you don't have a "raw integer literal" anyway, you can match on `OfNat.ofNat (nat_lit n)` and `-OfNat.ofNat (nat_lit n)`